frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-topology/utils.ts

```diff
@@ -297,6 +297,9 @@ export const connectFinallyTasksToNodes = (
   const finallyTasks = pipelineRun
     ? getFinallyTasksWithStatus(pipeline, pipelineRun)
     : pipeline.spec.finally;
```

Comment:
```suggestion
    : pipeline.spec.finally || [];
```
With this change, we can remove the previous check 3 lines above this code.